### PR TITLE
trace/dns: add timestamp field

### PR DESCRIPTION
**Pull request by wedaly**
# trace/dns: add timestamp field

Add a timestamp field to DNS trace events so we can see when the DNS packet was observed. The timestamp comes from bpf_ktime_get_ns(), which reports the time in nanoseconds since system boot, excluding suspend time (CLOCK_MONOTONIC).

Related issue: https://github.com/inspektor-gadget/inspektor-gadget/issues/1196

## How to use

Follow the steps in https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/guides/trace/dns.md

## Testing done
Tested in minikube by running `./kubectl-gadget trace dns -n demo -o custom-columns=timestamp,qtype,type,name`

image

### CODE REVIEW

File: pkg/gadgets/trace/dns/types/dns.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type Event struct {
 	PktType    string     `json:"pktType,omitempty" column:"type,minWidth:7,maxWidth:9"`
 	QType      string     `json:"qtype,omitempty" column:"qtype,minWidth:5,maxWidth:10"`
 	DNSName    string     `json:"name,omitempty" column:"name,width:30"`
+	Timestamp  uint64     `json:"timestamp,omitempty" column:"timestamp"`
```
Comment (wedaly) [inline pkg/gadgets/trace/dns/types/dns.go:38]: I wasn't sure what other properties to set for `column` here. I see that [traceloop sets `width:7,fixed,hide`](https://github.com/inspektor-gadget/inspektor-gadget/blob/e3546b642cefcbb5df31b11e3592219c0118facd/pkg/gadgets/traceloop/types/types.go#L34), but that truncated the timestamps. Thoughts?

File: pkg/gadgets/trace/dns/tracer/bpf/dns.c [hunk 2]
```diff
@@ -105,6 +105,7 @@ int ig_trace_dns(struct __sk_buff *skb)
 		return 0;
 
 	struct event_t event = {0,};
+	event.timestamp = bpf_ktime_get_ns();
```
Comment (wedaly) [inline pkg/gadgets/trace/dns/tracer/bpf/dns.c:108]: Is this the right helper to use? Wasn't sure if `bpf_ktime_get_coarse_ns` or `bpf_ktime_get_boot_ns` might be more appropriate.

**Comment by wedaly** [comment]
looks like this overlaps with https://github.com/inspektor-gadget/inspektor-gadget/pull/1205

**Comment by alban** [comment]
Hi @wedaly, sorry for the duplicate work.
#1205 does not implement everything, for example it does not show the duration between a dns request and the dns response. Would you like to work on that once #1205 is merged? I am going to post some thoughts on #1196.

**Comment by wedaly** [comment]
> Hi @wedaly, sorry for the duplicate work. #1205 does not implement everything, for example it does not show the duration between a dns request and the dns response. Would you like to work on that once #1205 is merged? I am going to post some thoughts on #1196.

No problem! Yes, I'd like to work on #1196. Thanks!

**Pull request closed without merging**